### PR TITLE
feat(backend): Move and List func added

### LINKED ISSUE #405: Storage backend: Implement MoveObject and ListObject

### PR DESCRIPTION
**Pull request by QuentinEscudierScalingo**
Fix #405

- [x] Add a changelog entry in `CHANGELOG.md`

### CODE REVIEW

File: storage/swift.go [hunk 1]
```diff
@@ -127,6 +127,22 @@ func (s *Swift) Info(ctx context.Context, path string) (types.Info, error) {
 	}, nil
 }
 
+func (s *Swift) Move(ctx context.Context, srcPath, dstPath string) error {
+	return s.conn.ObjectMove(ctx, s.cfg.Container, srcPath, s.cfg.Container, dstPath)
```
Comment (EtienneM) [inline storage/swift.go:131]: I prefer we annotate the errors in this lib
```suggestion
	err := s.conn.ObjectMove(ctx, s.cfg.Container, srcPath, s.cfg.Container, dstPath)
	if err != nil {
		return errors.Wrapf(err, "fail to move the object")
	}
	return nil
```

File: storage/s3.go [hunk 2]
```diff
@@ -219,6 +222,41 @@ func (s *S3) Info(ctx context.Context, path string) (types.Info, error) {
 	return info, nil
 }
 
+func (s *S3) Move(ctx context.Context, srcPath, dstPath string) error {
+	dstPath = fullPath(dstPath)
+	srcPathWithBucket := fullPath(srcPath)
+	srcPathWithBucket = fmt.Sprintf("%s/%s", s.cfg.Bucket, srcPathWithBucket)
```
Comment (EtienneM) [inline storage/s3.go:228]:
```suggestion
	srcPathWithBucket := fmt.Sprintf("%s/%s", s.cfg.Bucket, fullPath(srcPath)
)
```

File: storage/s3.go [hunk 3]
```diff
@@ -219,6 +222,41 @@ func (s *S3) Info(ctx context.Context, path string) (types.Info, error) {
...
+		Bucket:     &s.cfg.Bucket,
+		Key:        &dstPath,
+		CopySource: &srcPathWithBucket,
+	}
+	_, err := s.s3client.CopyObject(ctx, copyInput)
+	if err != nil {
+		return errors.Wrapf(err, "fail to copy object '%v' to '%v'", srcPathWithBucket, dstPath)
+	}
+
+	err = s.Delete(ctx, srcPath)
```
Comment (EtienneM) [inline storage/s3.go:239]: Please wrap this error

File: storage/s3.go [hunk 4]
```diff
@@ -219,6 +222,41 @@ func (s *S3) Info(ctx context.Context, path string) (types.Info, error) {
 	return info, nil
 }
 
+func (s *S3) Move(ctx context.Context, srcPath, dstPath string) error {
```
Comment (EtienneM) [inline storage/s3.go:225]: No test for this function?

File: storage/s3.go [hunk 5]
```diff
@@ -219,6 +222,41 @@ func (s *S3) Info(ctx context.Context, path string) (types.Info, error) {
...
+	_, err := s.s3client.CopyObject(ctx, copyInput)
+	if err != nil {
+		return errors.Wrapf(err, "fail to copy object '%v' to '%v'", srcPathWithBucket, dstPath)
+	}
+
+	err = s.Delete(ctx, srcPath)
+	return err
+}
+
+func (s *S3) List(ctx context.Context, prefix string) ([]string, error) {
```
Comment (EtienneM) [inline storage/s3.go:243]: No test for this function?

**Review by EtienneM — CHANGES_REQUESTED** [review]

File: storage/swift.go [hunk 6]
```diff
@@ -137,12 +137,11 @@ func (s *Swift) Move(ctx context.Context, srcPath, dstPath string) error {
...
 func (s *Swift) List(ctx context.Context, prefix string) ([]string, error) {
-	listInput := &swift.ObjectsOpts{
+	objects, err := s.conn.ObjectNames(ctx, s.cfg.Container, &swift.ObjectsOpts{
 		Prefix: prefix,
-	}
-	objects, err := s.conn.ObjectNames(ctx, s.cfg.Container, listInput)
+	})
 	if err != nil {
-		return nil, errors.Wrap(err, "fail to list objects")
+		return nil, errors.Wrap(err, "fail to list objects in '%v'", prefix)
```
Comment (QuentinEscudierScalingo) [inline storage/swift.go:144]: maybe you wanted to `errors.Wrapf` ?

Comment (QuentinEscudierScalingo) [reply storage/swift.go:144]: otherwise LGTM

**Review by EtienneM — APPROVED** [review]

File: storage/swift.go (hunk 6 above)
Comment (EtienneM) [reply storage/swift.go]: Nice catch 👍🏾

**Pull request merged**